### PR TITLE
Stress test by submiting the requests

### PR DESCRIPTION
**Pull request by moyodiallo**

### CODE REVIEW

File: stress/stress_submit.ml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+let ocaml_version = "4.14.0"
+let opam_repository_commit = "34576e67c88137d40ce6ff9e252d549e9e87205f"
+
+let make_requests limit =
+  let* vars = Utils.get_vars ~ocaml_package_name ~ocaml_version () in
+  let* opam_packages = Utils.get_opam_packages () in
+  let rec requests packages acc n =
+    match packages with
+    | [] -> Lwt.return acc
+    | package :: tl ->
```
Comment (MisterDA) [inline stress/stress_submit.ml:39]: Shouldn't this rather be a fold over the packages list? Maybe with `Lwt_list.fold_left_s`?
If not, you should reorder the arguments to 
```ocaml
  let rec requests acc n = function
    | [] -> Lwt.return acc
    | package :: tl ->
```

File: stress/utils.ml [hunk 2]
```diff
@@ -36,3 +36,10 @@ let get_vars ~ocaml_package_name ~ocaml_version ?arch () =
 
 let get_opam_file pv =
   Solver_service.Process.pread ("", [| "opam"; "show"; "--raw"; pv |])
+
+let get_opam_packages () =
+  let open Lwt.Infix in
+  Solver_service.Process.pread
+    ("", [| "opam"; "list"; "--short"; "--color"; "never" |])
+  >|= (fun s -> String.split_on_char '\n' s)
+  >|= List.map (fun p -> String.trim p)
```
Comment (MisterDA) [inline stress/utils.ml:45]: Is that necessary?

**Review by MisterDA — CHANGES_REQUESTED** [review]

File: stress/stress_submit.ml (hunk 1 above)
Comment (moyodiallo) [reply stress/stress_submit.ml:39]: Cool thanks.

File: stress/utils.ml [hunk 3]
```diff
@@ -36,3 +36,10 @@ let get_vars ~ocaml_package_name ~ocaml_version ?arch () =
 
 let get_opam_file pv =
   Solver_service.Process.pread ("", [| "opam"; "show"; "--raw"; pv |])
+
+let get_opam_packages () =
+  let open Lwt.Infix in
+  Solver_service.Process.pread
+    ("", [| "opam"; "list"; "--short"; "--color"; "never" |])
```
[the remark this answers is not in the document]
Comment (moyodiallo) [reply stress/utils.ml:43]: Sound good.

File: stress/README.md [hunk 4]
```diff
@@ -10,4 +10,12 @@ This should give you a capnp address to copy. Then run the stress test passing i
 
 ```
 $ dune exec -- stress/stress.exe capnp://...
-```
\ No newline at end of file
+```
+
+To submit the stress tests to a scheduler that handle a solver-worker, the scheduler could be a
+remote. by the default the number of request is limited to 30 (--limit). Make sure, the
+cache of the solver-worker is reset.
```
[the remark this answers is not in the document]
Comment (moyodiallo) [reply stress/README.md:17]: Sound good.

During the start of `solver-worker` we use the parameter `--state-dir` like `--state-dir=var`, and reseting the cache is doing `rm -r var/solver` after restart the `solver-worker`.

File: stress/utils.ml (hunk 3 above)
[the remark this answers is not in the document]
Comment (moyodiallo) [reply stress/utils.ml:43]: Yeah `--short` exclude all comments.

File: stress/utils.ml (hunk 2 above)
Comment (maiste) [reply stress/utils.ml:45]: In case you keep it, it can be simplified:
```suggestion
  >|= String.split_on_char '\n'
  >|= List.map String.trim
```

**Pull request merged**